File: Method3/PrimeNumbers.cpp
#include<iostream>
using namespace std;

int isPrime(int num) { // Time Complexity O(sqrt(N))
  if (num < 2) {
    return 0;
  } else {
    for (int i = 2; i * i <= num; i++) { // or for (int i = 2; i <= sqrt(num); i++)
      if (num % i == 0) {
        return 0;
      }
    }
    return 1;
  }
}

int main() {
  int num;

  cout << "Enter Value of n... ";
  cin >> num;

  if (isPrime(num)) {
    cout << num << " is a Prime Number.";
  } else {
    cout << num << " is not a Prime Number.";
  }
  return 0;
}

File: Method3/PrimeNumbers.py
def isPrime(num): # Time Complexity O(sqrt(N))
  if num < 2:
    return False;
  else:
    i = 2
    while (i * i <= num):
      if num % i == 0:
        return False;
      i += 1;
    return True;

num = int(input('Enter Value of n... '))
if isPrime(num):
  print(num, 'is a Prime Number.')
else:
  print(num,'is not a Prime Number.')